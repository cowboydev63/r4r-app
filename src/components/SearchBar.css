.searchbar__container {
    display: flex;
    width: 700px;
    max-width: 100%;
}

.searchbar__container.r4r-DEFAULT {
    height: 100%;
    flex: 1;
}

.searchbar__container.r4r-DEFAULT input[type="text"] {
    border: none;
    padding: 6px 10px;
    font-size: 18px;
    color: #333;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.16), 0 0 0 1px rgba(0,0,0,0.08);
}

.searchbar__container .searchbar__button--submit {
    margin: 0;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    background: #2a71a5;
}

.searchbar__container .searchbar__button--submit:hover {
    background: #266594;
}

.searchbar__container .searchbar__button--submit.r4r-DEFAULT {
    padding: 6px;
    color: white;
    background: #169bd5;
    border-radius: 7px;
    font-size: 22px;
    box-shadow: 0 12px 5px -10px rgba(0,0,0,0.16);
    cursor: pointer;
    transition: box-shadow 125ms linear;
}

.searchbar__container .searchbar__button--submit.r4r-DEFAULT:hover {
    box-shadow: 0 18px 15px -15px rgba(0,0,0,0.2);
}

.searchbar__container .searchbar__button--submit.r4r-DEFAULT:active {
    box-shadow: 0 10px 5px -8px rgba(0,0,0,0.2);
}