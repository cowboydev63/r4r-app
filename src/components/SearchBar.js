import React from 'react';
import PropTypes from 'prop-types';
import { Theme } from '../theme';
import './SearchBar.css';

class SearchBar extends React.PureComponent {
    static propTypes = {
        page: PropTypes.oneOf(['home', 'resource', 'results']),
        value: PropTypes.string,
        onChange: PropTypes.func.isRequired,
        onSubmit: PropTypes.func.isRequired,
        placeholder: PropTypes.string,
    }

    static defaultProps = {
        onChange: () => {},
        onSubmit: () => {},
        placeholder: '',
    }

    onChange = e => {
        e.preventDefault();
        this.props.onChange({
            page: this.props.page,
            value: e.target.value,
        })
    }

    onSubmit = e => {
        e.preventDefault();
        this.props.onSubmit();
    }

    render() {
        return(
            <Theme element="form" onSubmit={ this.onSubmit } className='searchbar__container'>
                <input 
                    type="text" 
                    value={ this.props.value }
                    onChange={ this.onChange }
                    aria-label="search"
                    placeholder={ this.props.placeholder }
                />
                <Theme
                    element="input" 
                    type="submit" 
                    value="Q"
                    className="searchbar__button--submit"
                />
            </Theme>
        )
    }
}

export default SearchBar;