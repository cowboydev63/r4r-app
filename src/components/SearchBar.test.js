import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from './SearchBar';

describe('SearchBar Component', () => {
  it('renders without crashing', () => {
    shallow(<SearchBar />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<SearchBar />);
    expect(wrapper).toMatchSnapshot();
  })

  it('executes searches on submit', () => {
    const mockFn = jest.fn()
    const wrapper = shallow(<SearchBar onSubmit={mockFn} />);
    wrapper.find('.searchbar__container').simulate('submit', { preventDefault() {} });
    expect(mockFn.mock.calls.length).toBe(1);
  })

  //TODO: When the issue with finding 'input is resolved, test onChange function
})