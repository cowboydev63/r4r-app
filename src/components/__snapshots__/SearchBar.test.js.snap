// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component renders correctly 1`] = `
<Theme
  className="searchbar__container"
  element="form"
  onSubmit={[Function]}
>
  <input
    aria-label="Input search text here to search for a resource"
    onChange={[Function]}
    placeholder=""
    type="text"
  />
  <Theme
    aria-label="submit search"
    className="searchbar__button--submit"
    element="button"
    type="submit"
  >
    <SVG
      className="r4r-svg-container"
      iconType="magnify"
    />
  </Theme>
</Theme>
`;
